timeseries/src/autogluon/timeseries/models/ensemble/array_based/abstract.py

```diff
@@ -0,0 +1,209 @@
+        The isotonization method to use (i.e. the algorithm to prevent quantile non-crossing).
+        Currently only "sort" is supported.
+    detect_and_ignore_failures: bool, default = True
+        Whether to detect and ignore "failed models", defined as models which have a loss that is larger
+        than 10x the median loss of all the models. This can be very important for the regression-based
+        ensembles, as moving the weight from such a "failed model" to zero can require a long training
+        time.
+    """
+
+    _regressor_type: Type[EnsembleRegressor]
+
+    def __init__(
+        self,
+        path: Optional[str] = None,
+        name: Optional[str] = None,
+        hyperparameters: Optional[dict[str, Any]] = None,
+        freq: Optional[str] = None,
+        prediction_length: int = 1,
+        covariate_metadata: Optional[CovariateMetadata] = None,
+        target: str = "target",
+        quantile_levels: Sequence[float] = (0.1, 0.2, 0.3, 0.4, 0.5, 0.6, 0.7, 0.8, 0.9),
+        eval_metric: Union[str, TimeSeriesScorer, None] = None,
+    ):
+        super().__init__(
+            path=path,
+            name=name,
+            hyperparameters=hyperparameters,
+            freq=freq,
+            prediction_length=prediction_length,
+            covariate_metadata=covariate_metadata,
+            target=target,
+            quantile_levels=quantile_levels,
+            eval_metric=eval_metric,
+        )
+        self.ensemble_regressor: Optional[EnsembleRegressor] = None
+        self._model_names: list[str] = []
+
+    def _get_default_hyperparameters(self) -> dict[str, Any]:
+        return {
+            "isotonization": "sort",
+            "detect_and_ignore_failures": True,
+        }
+
+    @staticmethod
+    def to_array(df: TimeSeriesDataFrame) -> np.ndarray:
+        """Given a TimeSeriesDataFrame object, or a list or dict of such objects, return
+        a single array composing the values contained in the data frames.
+
+        Parameters
+        ----------
+        df
+            TimeSeriesDataFrame to convert to an array.
+
+        Returns
+        -------
+        array
+            of shape (items, nr_timesteps, quantiles).
+        """
+        df = df.sort_index()
+        array = df.values
+        nr_items = len(df.index.get_level_values(0).unique())
+        shape = (
+            nr_items,  # nr_items
+            df.shape[0] // nr_items,  # timesteps per item
+            df.shape[1],  # number of quantiles
+        )
+        return array.reshape(shape)
+
+    def _split_data_per_window(
+        self,
+        data_per_window: list[TimeSeriesDataFrame],
+    ):
+        """Split the given `data_per_window` into ground truth for that window (fold) and the past data."""
+        past_data_per_window = [y.slice_by_timestep(None, -self.prediction_length) for y in data_per_window]
+        ground_truth_per_window = [y.slice_by_timestep(-self.prediction_length, None) for y in data_per_window]
```

Comment:
Should we have some check that ensures that enough observations are available for each item? This logic could also live in the trainer when we get the data for fitting the ensemble.